src/components/pages/UserLibraryAccessPage.tsx

```diff
@@ -0,0 +1,334 @@
+        chkEnableAllFolders.checked = user.Policy.EnableAllFolders;
+        triggerChange(chkEnableAllFolders);
+    }, []);
+
+    const loadChannels = useCallback((user, channels) => {
+        const itemsArr: ItemsArr[] = [];
+
+        for (const folder of channels) {
+            console.log('EnableAllChannels', user.Policy.EnableAllChannels);
+            const isChecked = user.Policy.EnableAllChannels || user.Policy.EnabledChannels.indexOf(folder.Id) != -1;
+            const checkedAttribute = isChecked ? ' checked="checked"' : '';
+            itemsArr.push({
+                Id: folder.Id,
+                Name: folder.Name,
+                checkedAttribute: checkedAttribute
+            });
+        }
+
+        setChannelsItems(itemsArr);
+
+        if (channels.length) {
+            element?.current?.querySelector('.channelAccessContainer').classList.remove('hide');
+        } else {
+            element?.current?.querySelector('.channelAccessContainer').classList.add('hide');
+        }
+
+        const chkEnableAllChannels = element.current.querySelector('.chkEnableAllChannels');
+        chkEnableAllChannels.checked = user.Policy.EnableAllChannels;
+        triggerChange(chkEnableAllChannels);
+    }, []);
+
+    const loadDevices = useCallback((user, devices) => {
+        const itemsArr: ItemsArr[] = [];
+
+        for (const device of devices) {
+            console.log('EnableAllDevices', user.Policy.EnableAllDevices);
+            const isChecked = user.Policy.EnableAllDevices || user.Policy.EnabledDevices.indexOf(device.Id) != -1;
+            const checkedAttribute = isChecked ? ' checked="checked"' : '';
+            itemsArr.push({
+                Id: device.Id,
+                Name: device.Name,
+                AppName : device.AppName,
+                checkedAttribute: checkedAttribute
+            });
+        }
+
+        setDevicesItems(itemsArr);
+
+        const chkEnableAllDevices = element.current.querySelector('.chkEnableAllDevices');
+        chkEnableAllDevices.checked = user.Policy.EnableAllDevices;
+        triggerChange(chkEnableAllDevices);
+
+        if (user.Policy.IsAdministrator) {
+            element?.current?.querySelector('.deviceAccessContainer').classList.add('hide');
+        } else {
+            element?.current?.querySelector('.deviceAccessContainer').classList.remove('hide');
+        }
+    }, []);
+
+    const loadUser = useCallback((user, mediaFolders, channels, devices) => {
+        setUserName(user.Name);
+        libraryMenu.setTitle(user.Name);
+        loadChannels(user, channels);
+        loadMediaFolders(user, mediaFolders);
+        loadDevices(user, devices);
+        loading.hide();
+    }, [loadChannels, loadDevices, loadMediaFolders]);
+
+    const loadData = useCallback(() => {
+        loading.show();
+        const userId = appRouter.param('userId');
+        // eslint-disable-next-line compat/compat
```

Comment:
Is this required to use `Promise`?

Reply:
previously was used  jQuery $.deferred
https://github.com/jellyfin/jellyfin-web/blob/7ea8ce7debae59cb91c8f1f3a647931f49217c93/src/controllers/dashboard/users/userlibraryaccess.js#L165-L173

Reply:
Ah I see this is an issue with our eslint configuration. The polyfills are not being included for jsx/tsx files. That probably deserves a separate PR. :smile: